src/Illuminate/Console/Command.php

```diff
@@ -370,15 +370,15 @@ public function anticipate($question, array $choices, $default = null)
      * Prompt the user for input with auto completion.
      *
      * @param  string  $question
-     * @param  array   $choices
+     * @param  array|callable $choices
      * @param  string|null  $default
      * @return mixed
      */
-    public function askWithCompletion($question, array $choices, $default = null)
+    public function askWithCompletion($question, $choices, $default = null)
     {
         $question = new Question($question, $default);
 
-        $question->setAutocompleterValues($choices);
+        is_callable($choices) ? $question->setAutocompleterCallback($choices) : $question->setAutocompleterValues($choices);
```

Comment:
Maybe use an if/else statement here which would look nicer and not make the line so long.

Reply:
You can also break the line (option I think more elegant)
`
is_callable($choices)
    ? $question->setAutocompleterCallback($choices)
    : $question->setAutocompleterValues($choices);
`

Reply:
Thanks @driesvints, I agree that the line is too long.

I like @danilopinotti's solution, I'll make that change.